### ISSUE TITLE
trying to get this working on HOME ASSISTANT core {python}

### ISSUE THREAD

**Issue by gatd** [tsightler/ring-mqtt#331]
mac mini mojave home assistant core python

installed via git clone
after running npm install i get
5 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force
after running  npm audit fix --force i get
# npm audit report

trim-newlines  <3.0.1
Severity: high
Regular Expression Denial of Service in trim-newlines - https://github.com/advisories/GHSA-7p7h-4mm5-852v
fix available via `npm audit fix --force`
Will install ring-client-api@4.2.3, which is a breaking change
node_modules/trim-newlines
  meow  3.4.0 - 5.0.0
  Depends on vulnerable versions of trim-newlines
  Depends on vulnerable versions of yargs-parser
  node_modules/meow
    stun  >=2.0.0
    Depends on vulnerable versions of meow
    node_modules/stun
      ring-client-api  4.3.0 - 9.25.0-beta.1
      Depends on vulnerable versions of stun
      node_modules/ring-client-api

yargs-parser  6.0.0 - 13.1.1
Severity: moderate
Prototype Pollution in yargs-parser - https://github.com/advisories/GHSA-p9pc-299p-vxgp

 if i run
ring-mqtt.js i get

Error: listen EADDRINUSE: address already in use :::55123
    at Server.setupListenHandle [as _listen2] (node:net:1331:16)
    at listenInCluster (node:net:1379:12)
    at Server.listen (node:net:1467:7)
    at Function.listen (/opt/ring-mqtt/node_modules/express/lib/application.js:618:24)
    at TokenApp.start (/opt/ring-mqtt/lib/tokenapp.js:39:34)
    at main (/opt/ring-mqtt/ring-mqtt.js:560:26) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::',
  port: 55123
}

any help with this please
thanks

**Comment by tsightler** [comment]
Don't try to fix the vulnerabilities as it will not work.  Unfortunately, ring-client-api is dependent on a few packages that haven't been maintained in a while, but these vulnerabilities do not pose a significant risk in the context of this project.

> Error: listen EADDRINUSE: address already in use :::55123

So something is already listening on the port 55123.  No idea what it is but that port is used by the web based token generator so you can either find out and stop whatever is conflicting, or manually generate a token using the alternative method.

**Comment by gatd** [comment]
re installed without fixing the vulnerabilities but still no go
is it possible for a step by step to get this running it would also help many other people?
thanks

**Comment by tsightler** [comment]
The instructions you have already followed are the step-by-step instructions.   As I've already pointed out, the error above clearly indicates that there is already a service using port 55123.  You must find and stop this service or use an alternative method to acquire the token as covered in the documentation.

I consider standard installs, which you have chosen to use, largely self-supported.  Installing the software manually this way assumes a basic level of system administration skill and understanding, if you don't have these skills, then the Docker method may be an easier path.  If you are not familiar with either of these methods, then perhaps this project isn't for you.

Closing as this is not a bug with this project and I don't provide support for basic systems administration.
